balancer/pickfirst/pickfirstleaf/pickfirstleaf.go

```diff
@@ -288,9 +299,18 @@ func (b *pickfirstBalancer) ExitIdle() {
 	b.mu.Lock()
 	defer b.mu.Unlock()
 	if b.state == connectivity.Idle && b.addressList.currentAddress() == b.addressList.first() {
```

Comment:
I initially added it because I was (incorrectly) thinking that ExitIdle will start the iteration over the address list from the start. This would cause the first subchannel to be connected which isn't in TF. If it was called multiple times, subchannels that were in TF earlier could complete the backoff and exit TF.

However, that's not the case. The address list is reset while entering IDLE, not exiting IDLE. So repeated calls to `requestConnectionLocked()` will be a no-op.

Thanks for spotting this, I've removed the check and the `addressList.first()` method  I added to support this.